### ISSUE TITLE
Three derivations might not respect existing instances in scope

### ISSUE THREAD

**Issue by kailuowang** [typelevel/kittens#129]
Namely 
```
object pure extends MkPureDerivation
object foldable extends MkFoldableDerivation
object monoid extends MkMonoidDerivation
```
These three are implemented differently from the other derivations. They probably don't respect existing instances.

**Comment by joroKr21** [comment]
The problem is that there are other inconsistencies as well. I will write up an issue / proposal what to do.

**Comment by joroKr21** [comment]
Fixed by #151, #139 and #132
